openc3/bin/openc3cli

```diff
@@ -648,6 +660,168 @@ def run_bridge(filename, params)
   end
 end
 
+def script_monitor(script_id)
+  ret_code = ERROR_CODE
+  require 'openc3/script'
+  OpenC3::RunningScriptWebSocketApi.new(id: script_id) do |api|
+    while (resp = api.read) do
+      # see ScriptRunner.vue for types and states
+      case resp['type']
+      when 'running', 'breakpoint', 'waiting'
+        puts resp.pretty_inspect
+      when 'error', 'fatal'
+        $script_interrupt_text = ''
+        puts 'script failed'
+        break
+      when 'file'
+        puts "Filename #{resp['filename']} scope #{resp['scope']}"
+      when 'line'
+        fn = resp['filename'].nil? ? '<no file>' : resp['filename']
+        puts "At [#{fn}:#{resp['line_no']}] state [#{resp['state']}]"
+        if resp['state'] == 'error'
+          $script_interrupt_text = ''
+          puts 'script failed'
+          break
+        end
+      when 'output'
+        puts resp['line']
+      when 'paused'
+        if resp['state'] == 'fatal'
+          $script_interrupt_text = ''
+          puts 'script failed'
+          break
+        end
+      when 'time'
+        puts resp.pretty_inspect
+      when 'complete'
+        $script_interrupt_text = ''
+        puts 'script complete'
+        ret_code = 0
+        break
+      else
+        puts resp.pretty_inspect
+      end
+    end
+  end
+  return ret_code
+end
+
+def cli_script_list(args=[])
+  path = ''
+  if (args[0] && args[0][0] == '/')
+    path = (args.shift)[1..-1]+'/'
+  end
+  scope = args[0]
+  scope ||= 'DEFAULT'
+  require 'openc3/script'
+  script_list(scope: scope).each do |script_name|
+    puts (script_name) if script_name.start_with?(path)
+  end
+  return 0
+end
+
+def cli_script_run(disconnect=false, environment={}, args=[])
+  #  we are limiting the wait for status, not the script run time
+  #  we make 0 mean 'forever'
+  ret_code = ERROR_CODE
+  wait_limit = 0
+  if (i = args.index('--wait'))
+    begin
+      args.delete('--wait')
+      # pull out the flag
+      seconds = args[i]
+      wait_limit = Integer(seconds, 10) # only decimal, ignore leading 0
+      args.delete_at(i)
+      # and its value
+    rescue ArgumentError
+      abort("  --wait requires a number of seconds to wait, not [#{seconds}]")
+    end
+  end
+  abort("No script file provided") if args[0].nil?
+  scope = args[1]
+  scope ||= 'DEFAULT'
+  require 'openc3/script'
+  id = script_run(args[0], disconnect: disconnect, environment: environment, scope: scope) # could raise
+  $script_interrupt_text = "  Script #{args[0]} still running remotely.\n" # for Ctrl-C
+  if (wait_limit < 1) then
+    ret_code = script_monitor(id)
+  else
+    Timeout::timeout(wait_limit, nil, "--wait #{wait_limit} exceeded") do
+      ret_code = script_monitor(id)
+    rescue Timeout::ExitException, Timeout::Error => tmoexcp
+      # Timeout exceptions are also thrown by the Websocket API, so we check
+      if tmoexcp.message =~ /^--wait /
+        puts tmoexcp.message+", detaching from running script #{args[0]}"
```

Comment:
add spaces around + sign.

Reply:
Right, that was nearly invisible only days after I wrote it.